### PR TITLE
add delay support

### LINKED ISSUE #15: Add support for Delay task
We need to add support for task delay. 
Use cases:
1. Want to start a time after sometime
2. Want to avoid retrying the task immediately but instead run again after some delay

We should also support exponential backoff for continuous retries.

### PR DESCRIPTION
**Pull request by vedhavyas**
Fixes #15 

We need to support this so that we can do exponential backoff with retry support especially with PoW transaction confirmations

### CODE REVIEW

File: worker.go [hunk 1]
```diff
@@ -56,49 +56,61 @@ func (w *CeleryWorker) StartWorker() {
...
-					if err != nil {
-						log.Println(fmt.Errorf("failed to encode Task Message: %v", err))
-						continue
-					}
-
-					err = w.broker.SendCeleryMessage(getCeleryMessage(enc))
-					if err != nil {
-						log.Println(fmt.Errorf("failed to enqueue Task: %v", err))
-					}
+					task.Settings.ETA = time.Now().UTC().Add(defaultBackOff * time.Duration(task.Tries))
```
Comment (vedhavyas) [inline worker.go]: we will update ETA here.
with defaultBackoff as 5 seconds and max tries at 15, we will have 10 minutes before we the task tries reaches the max.

File: message.go [hunk 2]
```diff
@@ -9,20 +9,28 @@ import (
...
+	// defaultBackOff with current tries for back off
+	defaultBackOff = time.Second * 5
+)
 
 // TaskSettings can be passed to the task with specific overrides.
 type TaskSettings struct {
-	MaxTries int    `json:"max_tries"`
-	ETA      string `json:"eta"`
+	MaxTries uint      `json:"max_tries"`
+	ETA      time.Time `json:"eta"`
```
Comment (vimukthi-git) [inline message.go]: I think there is some confusion on `delay` vs `ETA`, 
- Delay is the delay for the task to be retried next
- ETA is the overall time a task is allowed to run.
So there needs to be two settings here one for Delay and one for ETA. Following is a visualisation of this
```
|------------------------------ETA--------------------------------------------------------------------------------------|
                                                                                    
|---delay time------|[------exec task (try 1)----]|----delay time----|[------exec task (try 2)----]|----delay time----| [ETA expired]
```

Comment (vedhavyas) [reply message.go]: Sure, i'll rename it to delay then 👍 
I dont see any use for ETA at the moment so

Comment (vimukthi-git) [reply message.go]: makes sense

**Pull request merged**